pew/pew.py

```diff
@@ -373,11 +373,23 @@ def setvirtualenvproject(env, project):
 
 
 def setproject_cmd():
-    """Given a virtualenv directory and a project directory, set the virtualenv up to be associated with the project."""
-    env = os.environ.get('VIRTUAL_ENV', args.get(1))
-    project = args.get(2, os.path.abspath('.'))
-    if not env:
+    """Given a virtualenv directory and a project directory, set the
+    virtualenv up to be associated with the project."""
+    if len(args) == 4:
+        env = args.get(2)
+        project = args.get(3)
+    elif len(args) == 3:
+        env = os.environ['VIRTUAL_ENV']
+        project = args.get(2)
+    elif len(args) == 2:
+        env = os.environ['VIRTUAL_ENV']
+        project = '.'
+
```

Comment:
I'm merging now the changes, and I just noticed that this whole segment is wrong (or, at least, doesn't preserve the same semantics as before), after adding this test:

    def check_project_dir(workon_home, env, projdir):
        return (workon_home / env / '.project').open().read() == projdir

    def test_setproject(workon_home, env1):
        del os.environ['VIRTUAL_ENV']
        with TemporaryDirectory() as tmpdir:
            res = invoke('setproject', 'env1', tmpdir)
            assert not res.err
            assert check_project_dir(workon_home, 'env1', tmpdir)

All of the indexes (2,3,4) and the corresponding lengths need to be decremented by 1